src/inmanta/deploy/state.py

```diff
@@ -210,66 +215,193 @@ def are_dependencies_compliant(self, resource: ResourceIdStr) -> bool:
 
         return all(self.resource_state[dep_id].status is ComplianceStatus.COMPLIANT for dep_id in dependencies)
 
-    def block_provides(self, resources: Set[ResourceIdStr]) -> Set[ResourceIdStr]:
+    def update_transitive_state(
+        self,
+        now_blocked: AbstractSet[ResourceIdStr],
+        verify_blocked: AbstractSet[ResourceIdStr],
+        verify_unblocked: set[ResourceIdStr],
+    ) -> Tuple[set[ResourceIdStr], set[ResourceIdStr]]:
         """
-        Marks the provides of the given resources as blocked.
 
-        Must be called under the scheduler lock. This method assumes that all the resources of the model version
-        are populated in the resources dictionary.
+        Update transitive states
 
-        :param resources: The set of resources for which the provides have to be blocked.
-        :return: The set of dependent resources that were newly marked as blocked (transitively).
+        :param now_blocked: resources that have become undefined
+        :param verify_blocked: resources that have added requires
+        :param verify_unblocked: resources that have lost requires relations
+
+        returns: <blocked, unlocked> resources
```

Comment:
This returns is in the wrong order. And I assume that we mean `unblocked` 
```suggestion
        returns: <unblocked, blocked> resources
```